src/test/java/com/rocketden/main/RoomTests.java

```diff
@@ -0,0 +1,121 @@
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.junit.Assert.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest(properties = "spring.datasource.type=com.zaxxer.hikari.HikariDataSource")
+@AutoConfigureMockMvc
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@Transactional
+public class RoomTests {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private static final String PUT_ROOM = "/api/v1/rooms";
+    private static final String POST_ROOM = "/api/v1/rooms";
+
+    @Test
+    public void joinNonExistentRoom() throws Exception {
+        // PUT request to join non-existent room should fail
+        JoinRoomRequest request = new JoinRoomRequest();
+        request.setRoomId("012345");
+        request.setPlayerName("Rocket");
+
+        JoinRoomResponse expected = new JoinRoomResponse();
+        expected.setMessage(JoinRoomResponse.ERROR_NOT_FOUND);
+
+        MvcResult result = this.mockMvc.perform(put(PUT_ROOM)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(request)))
+                .andDo(print()).andExpect(status().isNotFound())
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+        JoinRoomResponse actual = Utility.toObject(jsonResponse, JoinRoomResponse.class);
+
+        assertEquals(expected.getMessage(), actual.getMessage());
+    }
+
+    @Test
+    public void createValidRoom() throws Exception {
+        // POST request to create valid room should return successful response
+        CreateRoomRequest request = new CreateRoomRequest();
+
+        CreateRoomResponse expected = new CreateRoomResponse();
+        expected.setMessage(CreateRoomResponse.SUCCESS);
+
+        MvcResult result = this.mockMvc.perform(post(POST_ROOM)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(request)))
+                .andDo(print()).andExpect(status().isOk())
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+        CreateRoomResponse actual = Utility.toObject(jsonResponse, CreateRoomResponse.class);
+
+        assertEquals(expected.getMessage(), actual.getMessage());
+    }
+
+    @Test
+    public void createAndJoinRoom() throws Exception {
```

Comment:
Could you add more general comments to this? I feel like I can follow it, but it's a really long test so that addition would be helpful.

Reply:
Thanks for the additions 👍